Common/Models/Commands/Command.cs

```diff
@@ -4,6 +4,23 @@ namespace Microsoft.Azure.Devices.Applications.RemoteMonitoring.Common.Models.Co
 {
     public class Command
     {
+        /// <summary>
+        /// Serialziation deserialziation constructor.
+        /// </summary>
+        internal Command()
```

Comment:
I think you need this here: [JsonConstructor] 
Otherwise the deserializer defaults to the public command and fails